ampd/src/grpc/blockchain_service.rs

```diff
@@ -74,12 +76,13 @@ where
         self.msg_queue_client
             .clone()
             .enqueue(msg)
+            .map_err(Arc::new)
```

Comment:
why does it have to be an arc here? The last map_err just converts it back to a status

Reply:
from the `and_then` documentation: "The Error type of this future and the future returned by `f` have to match."  And `rx` uses arc, so I had to convert

Reply:
ah!